### ISSUE TITLE
Avoid exploding when incrementing flow control windows of nonexistent streams.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8682]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)| @Lukasa reported|
|-|-|
|Trac ID|trac#8682|
|Type|defect|
|Created|2016-07-26 08:32:46Z|
|Branch|https://github.com/twisted/twisted/tree/8682-lukasa-window-update-closed|

If a stream is closed when data is received, it's possible that we may attempt to increment the stream flow control window and be forbidden from doing so (PROTOCOL_ERROR). Right now this will kill the server. That's not good!

<details><summary>Searchable metadata</summary>

```
trac-id__8682 8682
type__defect defect
reporter__Lukasa Lukasa
priority__normal normal
milestone__None None
branch__8682_lukasa_window_update_closed 8682-lukasa-window-update-closed
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__None None
time__1469521966301446 1469521966301446
changetime__1470237998152574 1470237998152574
version__None None
owner__Lukasa Lukasa

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_1"></a>|@Lukasa commented|
|-|-|

In changeset 279ca63a4d78e544bac6656e915f0a416e929be8
```
#!CommitTicketReference repository="" revision="279ca63a4d78e544bac6656e915f0a416e929be8"
Failing test case for #8682.
```

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa commented|
|-|-|

In changeset fdce5344128a6d6caeb761b5c7c92f2f33992315
```
#!CommitTicketReference repository="" revision="fdce5344128a6d6caeb761b5c7c92f2f33992315"
Topfile for #8682.
```

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_3"></a>|@Lukasa commented|
|-|-|

Ok, patch available at: https://github.com/twisted/twisted/pull/455

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @Lukasa|
|-|-|

Thanks for the changes. They look good.

Please clarify why we still need/want to increment the connection window and then merge.

The other comments are minor.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub set status to `closed`|
|-|-|

In changeset 4ed10e691b0ba075736fb66008f0542ab67cee62
```
#!CommitTicketReference repository="" revision="4ed10e691b0ba075736fb66008f0542ab67cee62"
Merge 8682-lukasa-window-update-closed: Don't explode when the client resets a stream while it uploads data.

Author: Lukasa
Reviewer: adiroiban
Fixes: #8682
```
